pandas/core/internals.py

```diff
@@ -108,14 +109,15 @@ class Block(PandasObject):
     def __init__(self, values, placement, ndim=None, fastpath=False):
         if ndim is None:
             ndim = values.ndim
-        elif values.ndim != ndim:
+        elif self._validate_ndim and values.ndim != ndim:
```

Comment:
@jreback would be curious to hear your thoughts here. Needed this so that `ExtensionBlock` could inherit from `Block` and call `super().__init__`. We *could* avoid this by setting `.values`, `.mgr_locs`, `.ndim` directly in `ExtensionBlock.__init__`, but I think it's best practice to always call your parent's init.

Reply:
yes, see my above comment